rules/group.go

```diff
@@ -84,6 +98,11 @@ type Group struct {
 // DefaultEvalIterationFunc is the default implementation.
 type GroupEvalIterationFunc func(ctx context.Context, g *Group, evalTimestamp time.Time)
 
+// AlwaysTrueShouldEvaluateFunc will be used to determine if a rule group
+// needs to be evaluated or deactivated. Deactivating a rule group will
+// reset metrics, and the internal state of rules but not unload the rule group.
+type AlwaysTrueShouldEvaluateFunc func(ctx context.Context, g *Group, evalTimestamp time.Time, doneCh <-chan struct{}) bool
```

Comment:
I think in [this comment](https://github.com/prometheus/prometheus/pull/14142/files#r1624224129) the suggestion was to change the name of the default function, and not the type.

`DefaultEvaluateRuleGroupFunc` -> `AlwaysTrueShouldEvaluateFunc`